ioctl/newcmd/did/didgethash.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package did
+
+import (
+	"encoding/hex"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/iotexproject/iotex-core/ioctl"
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/newcmd/action"
+	"github.com/iotexproject/iotex-core/ioctl/util"
+)
+
+// Multi-language support
+var (
+	_getHashCmdUses = map[config.Language]string{
+		config.English: "gethash (CONTRACT_ADDRESS|ALIAS) DID",
+		config.Chinese: "gethash (合约地址|别名) DID",
+	}
+	_getHashCmdShorts = map[config.Language]string{
+		config.English: "Gethash get DID doc's hash on IoTeX blockchain",
+		config.Chinese: "Gethash 在IoTeX链上获取相应DID的doc hash",
+	}
+)
+
+// NewDidGetHash represents the did get hash command
+func NewDidGetHash(client ioctl.Client) *cobra.Command {
+	use, _ := client.SelectTranslation(_getHashCmdUses)
+	short, _ := client.SelectTranslation(_getHashCmdShorts)
+
+	cmd := &cobra.Command{
+		Use:   use,
+		Short: short,
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			contract, err := client.Address(args[0])
+			if err != nil {
+				return errors.Wrap(err, "failed to get contract address")
+			}
+			addr, err := address.FromString(contract)
+			if err != nil {
+				return errors.Wrap(err, "invalid contract address")
+			}
+
+			abi, err := abi.JSON(strings.NewReader(DIDABI))
+			if err != nil {
+				return errors.Wrap(err, "failed to read abi")
+			}
+			bytecode, err := encodeGet(abi, _getHashName, args[1])
+			if err != nil {
+				return errors.Wrap(err, "invalid bytecode")
+			}
+
+			result, err := action.Read(client, addr, "0", bytecode, contract, 100000)
```

Comment:
`100000` is gas limit default value?

Reply:
Yes, check [this](https://github.com/iotexproject/iotex-desktop-wallet/issues/381) for more info.

Reply:
The value is 
https://github.com/iotexproject/iotex-core/blob/49ce6130a40a48838ca4ffbc4757763b6931633a/ioctl/cmd/action/action.go#L52